lib/msf/core/exploit/cmd_stager.rb

```diff
@@ -93,14 +93,16 @@ def execute_cmdstager(opts = {})
 
       sent = 0
       total_bytes = 0
-      cmd_list.each { |cmd| total_bytes += cmd.length }
+      cmd_list.each { |cmd| total_bytes += cmd.bytesize }
 
       delay = opts[:delay]
       delay ||= 0.25
 
       cmd_list.each do |cmd|
+        # calculate string beforehand length in case exploit mutates string
+        command_length = cmd.bytesize
```

Comment:
Context: some of the `execute_command` implementations directly mutate the string with `gsub!` etc - which causes the incorrect string length/byte counting percentage calculation:

https://github.com/rapid7/metasploit-framework/blob/25e8cf8faf56059a6320e0e4df9c78e595f83293/modules/exploits/multi/ftp/pureftpd_bash_env_exec.rb#L101-L105

So we opt for counting the bytes before the module implementation intentionally/unintentionally mutates the command out of band